### PR TITLE
Prevent duplicate SPIR-V decorations

### PR DESCRIPTION
**Pull request by alan-baker**
* Update SPIRV-Tools
  * https://github.com/KhronosGroup/SPIRV-Tools/pull/5641 added validation that caught errors
* Modified glslang to prevent duplicate Restrict and Coherent decorations

### CODE REVIEW

File: Test/baseResults/hlsl.inf.vert.out [hunk 1]
```diff
@@ -136,7 +136,7 @@ Shader version: 500
               11:             TypePointer Function 6(float)
               13:    6(float) Constant 4286578688
               15:    6(float) Constant 2139095040
-              29:    6(float) Constant 4290772992
```
Comment (alan-baker) [inline Test/baseResults/hlsl.inf.vert.out:139]: This was failing on ToT main so I included it for a clean build.

Comment (alan-baker) [reply Test/baseResults/hlsl.inf.vert.out:139]: This fails on all the bots. Not sure why it passes locally.

**Review by alan-baker — COMMENTED** [review]
I missed a duplicate RelaxedPrecision, but I'm not certain where it's coming from.

File: Test/baseResults/hlsl.inf.vert.out (hunk 1 above)
Comment (alan-baker) [reply Test/baseResults/hlsl.inf.vert.out]: Reverted this change.

**Comment by alan-baker** [comment]
> I missed a duplicate RelaxedPrecision, but I'm not certain where it's coming from.

This is resolved. This PR is ready for review.

**Review by dneto0 — APPROVED** [review]

**Review by arcady-lunarg — APPROVED** [review]

**Pull request merged**